hub/saml-engine/src/integration-test/java/uk/gov/ida/integrationtest/hub/samlengine/apprule/support/ConfigStubRule.java

```diff
@@ -30,13 +44,35 @@ public void setupStubForCertificates(String issuer, String signingCertString, St
     }
 
     private void registerStubForCertificates(String issuer, CertificateDto signingCertificate, CertificateDto encryptionCertificate) throws JsonProcessingException {
-        String signingCertificateUri = UriBuilder.fromPath(Urls.ConfigUrls.SIGNATURE_VERIFICATION_CERTIFICATES_RESOURCE).buildFromEncoded(StringEncoding.urlEncode(issuer)).toASCIIString();
         Collection<CertificateDto> signingCertificates = new ArrayList<>();
         signingCertificates.add(signingCertificate);
-        register(signingCertificateUri, OK.getStatusCode(), signingCertificates);
+        stubFor(get(urlPathEqualTo(getPath(Urls.ConfigUrls.SIGNATURE_VERIFICATION_CERTIFICATES_RESOURCE, issuer)))
+                .willReturn(
+                        aResponse()
+                        .withStatus(200)
+                                .withHeader("Content-Type", APPLICATION_JSON)
+                        .withBody(objectMapper.writeValueAsString(signingCertificates)
+                        )
+                )
+        );
+
+        String encryptionCertificateUri = getPath(Urls.ConfigUrls.ENCRYPTION_CERTIFICATES_RESOURCE, issuer);
+        stubFor(get(urlPathEqualTo(encryptionCertificateUri))
+                .willReturn(
+                        aResponse()
+                                .withStatus(200)
+                                .withHeader("Content-Type", APPLICATION_JSON)
+                                .withBody(objectMapper.writeValueAsString(encryptionCertificate)
+                                )
+                )
+        );
+    }
 
-        String encryptionCertificateUri = UriBuilder.fromPath(Urls.ConfigUrls.ENCRYPTION_CERTIFICATES_RESOURCE).buildFromEncoded(StringEncoding.urlEncode(issuer)).toASCIIString();
-        register(encryptionCertificateUri, OK.getStatusCode(), encryptionCertificate);
+    private String getPath(String encryptionCertificatesResource, String param) {
+        return UriBuilder
+                .fromPath(encryptionCertificatesResource)
+                .buildFromEncoded( StringEncoding.urlEncode(param).replace("+", "%20"))
+                .toString();
     }
 
     public void setUpStubForShouldHubSignResponseMessagesForLegacySamlStandard(String issuerEntityId) throws JsonProcessingException {
```

Comment:
Can the name of this method be brought in line with the one below? That way it would be easier to see what differs in the setup between

`shouldGenerateRpAuthnResponseWithUnsignedMessage` and `shouldGenerateRpAuthnResponseWithMessageSignedByHubUsingSamlProfileStandard`

Reply:
should be updated now